org.eclipse.xtext.xbase.lib/src/org/eclipse/xtext/xbase/lib/MapExtensions.java

```diff
@@ -93,21 +106,316 @@ public boolean apply(Map.Entry<K, V> input) {
+	 * @param map the map to update.
+	 * @param key the key to remove.
+	 * @return the map with the content of the map except the key.
+	 * @since 2.14
+	 */
+	@Pure
+	public static <K, V> Map<K, V> operator_minus(Map<K, V> map, final K key) {
+		return Maps.filterKeys(map, new Predicate<K>() {
+			@Override
+			public boolean apply(K input) {
+				return !Objects.equal(input, key);
+			}
+		});
+	}
+
+	/**
+	 * Replies the elements of the left map without the pairs in the right map.
+	 * If the pair's values differ from
+	 * the value within the map, the map entry is not removed.
+	 *
+	 * <p>
+	 * The difference is an immutable
+	 * snapshot of the state of the maps at the time this method is called. It
+	 * will never change, even if the maps change at a later time.
+	 * </p>
+	 *
+	 * <p>
+	 * Since this method uses {@code HashMap} instances internally, the keys of
+	 * the supplied maps must be well-behaved with respect to
+	 * {@link Object#equals} and {@link Object#hashCode}.
+	 * </p>
+	 *
+	 * @param <K> type of the map keys.
+	 * @param <V> type of the map values.
+	 * @param left the map to update.
+	 * @param right the pairs to remove.
+	 * @return the map with the content of the left map except the pairs of the right map.
+	 * @since 2.14
+	 */
+	@Pure
+	public static <K, V> Map<K, V> operator_minus(Map<K, V> left, final Map<? extends K, ? extends V> right) {
+		return Maps.filterEntries(left, new Predicate<Entry<K, V>>() {
+			@Override
+			public boolean apply(Entry<K, V> input) {
+				final V value = right.get(input.getKey());
+				if (value == null) {
+					return input.getValue() == null && right.containsKey(input.getKey());
+				}
+				return !Objects.equal(input.getValue(), value);
+			}
+		});
+	}
+
+	/**
+	 * Replies the elements of the given map except the pairs with the given keys.
+	 *
+	 * <p>
+	 * The replied map is a view on the given map. It means that any change
+	 * in the original map is reflected to the result of this operation.
+	 * </p>
+	 *
+	 * @param <K> type of the map keys.
+	 * @param <V> type of the map values.
+	 * @param map the map to update.
+	 * @param keys the keys of the pairs to remove.
+	 * @return the map with the content of the map except the pairs.
+	 * @since 2.14
+	 */
+	@Pure
+	public static <K, V> Map<K, V> operator_minus(Map<K, V> map, final Iterable<?> keys) {
+		return Maps.filterKeys(map, new Predicate<K>() {
+			@Override
+			public boolean apply(K input) {
+				return !Iterables.contains(keys, input);
+			}
+		});
+	}
+
+	/**
+	 * Replies the value associated to the given key in the map.
+	 *
+	 * @param <K> type of the map keys.
+	 * @param <V> type of the map values.
+	 * @param map the map to consider.
+	 * @param key the key of the value.
+	 * @return the value associated to the key, or <code>null</code> if
+	 *     the key was not found.
+	 * @since 2.14
+	 */
+	@Pure
+	@Inline(value = "$1.get($2)")
+	public static <K, V> V operator_mappedTo(Map<K, V> map, K key) {
```

Comment:
This will break existing code, e.g. `null -> foo` now compile to a Pair but will become an attempt to invoke `((Map<..>)null).get(foo)`

Reply:
I will remove the inline annotation in order to not break the existing code.

Reply:
The problem is not about the inline, but about the overload of the `->` operator for the arguments `(Object, Object)`

Reply:
Ok. My mistake. I propose to remove the operator_mappedTo definition.